ambry-router/src/test/java/com.github.ambry.router/DeleteManagerTest.java

```diff
@@ -195,6 +195,42 @@ public void testBlobExpired() throws Exception {
     testWithErrorCodes(serverErrorCodes, partition, serverLayout, RouterErrorCode.BlobExpired, deleteErrorCodeChecker);
   }
 
+  /**
+   * Test the case when one server store responds with {@code Blob_Authorization_Failure}, and other servers
+   * respond with {@code Blob_Not_Found}. The delete operation should be able to resolve the
+   * router error code as {@code Blob_Authorization_Failure}. The order of received responses is the same as
+   * defined in {@code serverErrorCodes}.
+   */
+  @Test
+  public void testBlobAuthorizationFailure() throws Exception {
+    ServerErrorCode[] serverErrorCodes = new ServerErrorCode[9];
+    Arrays.fill(serverErrorCodes, ServerErrorCode.Blob_Not_Found);
+    serverErrorCodes[5] = ServerErrorCode.Blob_Authorization_Failure;
```

Comment:
Is there any specific reason we chose index `5` here to set `Blob_Authorization_Failure`?(If not, we might use a random number here to make it more general)

Reply:
This is related to `DELETE_PARALLELISM` which is 3 in this test. The index here should < 9 - DELETE_PARALLELISM to make sure we don't hit `Blob_Not_Found`. `testBlobExpired` is also using index 5.